### PR TITLE
Release/5.0.3

### PR DESCRIPTION
**Pull request by motin**
Collaborative PR to land 5.0.3 with clean-ups, smaller improvements and critical fixes without introducing any backwards incompatible changes to the API.

To use this release candidate in your add-on: `npm install mozilla/shield-studies-addon-utils#release/5.0.3 --save`

Restored CI tests with latest fx nightly versions (including fix for #228 via https://github.com/mozilla/shield-studies-addon-utils/pull/229)

### CODE REVIEW

File: examples/small-study/web-ext-config.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const defaultConfig = {
     overwriteDest: true,
   },
   run: {
-    firefox: "firefox",
+    firefox: "nightly",
```
Comment (gregglind) [inline examples/small-study/web-ext-config.js]: I am still really not a fan of this.  Studies mostly run on release.  But I will accept it.

File: test/functional/browser.study.api.js [hunk 2]
```diff
@@ -773,6 +773,113 @@ describe("PUBLIC API `browser.study` (not specific to any add-on background logi
       });
     });
 
+    describe("setup of an ineligible study should result in endStudy('ineligible') without even emitting onReady", function() {
```
Comment (gregglind) [inline test/functional/browser.study.api.js]: This is an excellent, hard test.

File: examples/small-study/src/studySetup.js [hunk 3]
```diff
@@ -42,9 +42,7 @@ const baseStudySetup = {
       ],
     },
     ineligible: {
-      baseUrls: [
-        "https://qsurvey.mozilla.com/s3/Shield-Study-Example-Survey/?reason=ineligible",
-      ],
+      baseUrls: [],
```
Comment (gregglind) [inline examples/small-study/src/studySetup.js]: We disagree here.  Sometimes it has been, to show a page saying "By the way, you actually don't match this", for opt-in studies.  I will make a note in docs though.

Comment (motin) [reply examples/small-study/src/studySetup.js]: Very true. I got blinded by mostly doing opt-out. Thanks for catching this!

File: examples/small-study/web-ext-config.js (hunk 1 above)
Comment (motin) [reply examples/small-study/web-ext-config.js]: Not me either, but we need to sort out one or both of these first before defaulting to branded/unbranded release again: https://github.com/mozilla/shield-studies-addon-utils/issues/241, https://github.com/mozilla/shield-studies-addon-utils/issues/242

**Pull request merged**